src/main/java/com/ss/mqtt/broker/network/MqttClient.java

```diff
@@ -26,9 +34,11 @@
 
     private volatile MqttVersion mqttVersion;
 
-    public MqttClient(@NotNull MqttConnection connection) {
+
+    public MqttClient(@NotNull MqttConnection connection, @NotNull SubscriptionService subscriptionService) {
         this.connection = connection;
         this.mqttVersion = MqttVersion.MQTT_5;
+        this.subscriptionService = subscriptionService;
```

Comment:
I think MqttClient should take reference of all services from its connection without storing it itself